docs/src/components/items/item.tsx

```diff
@@ -0,0 +1,350 @@
+import entries from 'object.entries';
+import SDKSupportTable from '../sdk-support-table/sdk-support-table';
+// import Icon from '@mapbox/mr-ui/icon';
+import Property from '../property.jsx';
+import Subtitle from '../subtitle.jsx';
+import {Markdown} from '~/components/markdown/markdown';
+
+interface IItem {
+    id: string;
+    name: string;
+    kind: string;
+    required: boolean;
+    minimum: number;
+    maximum: number;
+    values?: string[] | object;
+    default?: string | boolean | number |  string[] | object ;
+    requires: string[];
+    function: object;
+    transition: boolean;
+    example?: string  | number |  string[] | object ;
+    'sdk-support': object;
+    expression: object;
+    headingLevel?: '2' | '3';
+
+}
+
+export default function Item (props:IItem) {
+
+    function type(spec = props, plural = false) {
+        switch (spec.type) {
+            case null:
+            case '*':
+                return;
+            case 'light':
+                return (
```

Comment:
This is exactly what I would like to avoid - the need to change this code when changing the spec file.